src/common/com/intellij/plugins/haxe/ide/HaxeCompletionDefine.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2018 Ilya Malanin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intellij.plugins.haxe.ide;
+
+import org.jetbrains.annotations.NotNull;
+
+public class HaxeCompletionDefine extends HaxeCompletionItem {
+  public HaxeCompletionDefine(@NotNull String name, @NotNull String description) {
```

Comment:
Is the whole point of this class and HaxeCompletionMeta to add `@NotNull` checks to the constructors?  Because I'm not seeing any other use that is improved by using these new classes.  The types don't seem to be checked anywhere (no instanceof), so I don't understand why we need them.

Reply:
I've separated completion items for further purposes such as meta arguments completion. It was made just as a boilerplate for further work.

Reply:
Okay.